quint/src/cliCommands.ts

```diff
@@ -554,7 +554,36 @@ export async function verifySpec(prev: TypecheckedStage): Promise<CLIProcedure<V
   const args = verifying.args
   // TODO error handing for file reads and deserde
   const loadedConfig = args.apalacheConfig ? JSON.parse(readFileSync(args.apalacheConfig, 'utf-8')) : {}
-  const parsedSpec = jsonStringOfOutputStage(pickOutputStage(prev))
+
+  // Flatten modules, replacing instances, imports and exports with their definitions
+  const [analysisErrors, analysisOutput] = analyzeModules(verifying.table, verifying.modules)
+
+  if (analysisErrors.length > 0) {
+    const errorLocator = mkErrorMessage(verifying.sourceMap)
+    const errors = Array.from(analysisErrors, errorLocator)
+    return cliErr('verification failed', { ...verifying, errors })
+  }
+
+  const { flattenedModules, flattenedTable, flattenedAnalysis } = flattenModules(
+    verifying.modules,
+    verifying.table,
+    verifying.idGen,
+    verifying.sourceMap,
+    analysisOutput
+  )
```

Comment:
iiuc, you can give the `flattenModules` the `verifying` object as its last argument:

```suggestion
  const { flattenedModules, flattenedTable, flattenedAnalysis } = flattenModules(
    verifying.modules,
    verifying.table,
    verifying.idGen,
    verifying.sourceMap,
    verifying
  )
```

It'd be nice if `flattenModules` just took a (sub-record) or the `TypecheckedStage`, so unpacking the other fields here weren't needed. But that's nothing to fuss with here ofc. :)